api docs: document stream format

Luap99 opened the pull request:
Document the attach, exec and logs output stream format. We use the same format as docker.

Fixes #19280

<!--
Thanks for sending a pull request!

Please make sure you've read our contributing guidelines and how to submit a pull request (https://github.com/containers/podman/blob/main/CONTRIBUTING.md#submitting-pull-requests).

In case you're only changing docs, make sure to prefix the pull-request title with "[CI:DOCS]". That will prevent functional tests from running and save time and energy.

Finally, be sure to sign commits with your real name. Since by opening
a PR you already have commits, you can add signatures if needed with
something like `git commit -s --amend`.
-->

#### Does this PR introduce a user-facing change?

<!--
If no, just write `None` in the release-note block below. If yes, a release note
is required: Enter your extended release note in the block below. If the PR
requires additional action from users switching to the new release, include the
string "action required".

For more information on release notes, please follow the Kubernetes model:
https://git.k8s.io/community/contributors/guide/release-notes.md
-->

```release-note
None
```

vrothberg commented on pkg/api/server/register_containers.go:
```diff
@@ -526,7 +526,95 @@ func (s *APIServer) registerContainersHandlers(r *mux.Router) error {
+	//   ### Stream format
+	//
+	//   When the TTY setting is disabled for the container,
+	//   the HTTP Content-Type header is set to application/vnd.docker.multiplexed-stream
```
I am unable to find track down where this is being set in Podman.  I see it in vendor.

Luap99 replied:
Good catch, yes looks like we always set `application/vnd.docker.raw-stream` regardless of tty or not.
I guess nobody really cares about the headers, I will see if I can fix it in the code. Feels weird to me that we even use these docker headers in the libpod API but it is what it is.

Luap99 replied:
ah looks like docker just added this type with api 1.42, https://github.com/moby/moby/blob/331854a126674175dde199c6516174d15862cf1c/api/server/router/container/container_routes.go#L160-L162

vrothberg reviewed (commented):
The redundancy in the swagger docs is unfortunate but nothing we can solve in this PR.  So LGTM but I have one comment on the media type.

mheon commented on pkg/api/server/register_containers.go:
```diff
@@ -526,7 +526,95 @@ func (s *APIServer) registerContainersHandlers(r *mux.Router) error {
+	//   After the headers and two new lines, the TCP connection can now be used
+	//   for raw, bidirectional communication between the client and server.
+	//
+	//   To hint potential proxies about connection hijacking, the client
```
"To inform" instead of "To hint" maybe?

Luap99 replied:
Sure this was mostly copied from the docker docs, I just removed the docker daemon references.

mheon commented on pkg/api/server/register_containers.go:
```diff
@@ -526,7 +526,95 @@ func (s *APIServer) registerContainersHandlers(r *mux.Router) error {
+	//
+	//   The header contains the information which the stream writes (`stdout` or
+	//   `stderr`). It also contains the size of the associated frame encoded in
+	//   the last four bytes (`uint32`).
```
Should mention byte-order of the u32 here. Not clear if we're sending little or big endian in this.

mheon replied:
I see that this is mentioned a few paragraphs down... Maybe unify that paragraph into here?

mheon commented on pkg/api/server/register_containers.go:
```diff
@@ -1319,7 +1410,95 @@ func (s *APIServer) registerContainersHandlers(r *mux.Router) error {
 	// summary: Attach to a container
-	// description: Hijacks the connection to forward the container's standard streams to the client.
+	// description: |
+	//   Attach to a container to read its output or send it input. You can attach
```
Maybe just mention that the format is identical to Compat Attach?

Luap99 replied:
Properly better to do it the other way around? I mean I expect only a small number of people to looks at out compat docs. I would assume most people would use the docker docs or the libpod specific endpoint docs. SO having it here sounds better to me.

Luap99 commented:
PTAL again, I removed the duplicated docs and added code the return the new docker header. I decided to do for libpod endpoint as well as it will allow optimizations on the client side.

baude commented:
/lgtm
